java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S2053_java.html

```diff
@@ -0,0 +1,43 @@
+<p>In cryptography, "salt" is extra piece of data which is included in a hashing algorithm. It makes dictionary attacks more difficult. Using a
+cryptographic hash function without an unpredictable salt increases the likelihood that an attacker will be able to successfully guess a hashed value
+such as a password with a dictionary attack.</p>
+<p>This rule raises an issue when a hashing function which has been specifically designed for hashing sensitive data, such as PBKDF2, is used with a
+non-random, reused or too short salt value. It does not raise an issue on base hashing algorithms such as sha1 or md5 as these are often used for
+other purposes.</p>
+<h2>Recommended Secure Coding Practices</h2>
+<ul>
+  <li> Use hashing functions generating their own salt or generate a long random salt of at least 32 bytes. </li>
+  <li> The salt is at least as long as the resulting hash value. </li>
+  <li> Provide the salt to a safe hashing function such as PBKDF2. </li>
+  <li> Save both the salt and the hashed value in the relevant database record; during future validation operations, the salt and hash can then be
+  retrieved from the database. The hash is recalculated with the stored salt and the value being validated, and the result compared to the stored
+  hash. </li>
+</ul>
+<h2>Noncompliant Code Example</h2>
```

Comment:
We should add an example with `PBEKeySpec` to show in the description that we also support this library.